### PR TITLE
OP-20633, OP-21175: feature implementing  the  user logged in events API's and Publish user details as events to rabbitmq.

### PR DESCRIPTION
**Pull request by sudhakaropsmx**
https://devopsmx.atlassian.net/browse/OP-20633
https://devopsmx.atlassian.net/browse/OP-21175

### CODE REVIEW

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/AuthenticationAuditListener.java [hunk 1]
```diff
@@ -88,7 +104,23 @@ private void handleAuthenticationEvent(
...
+  private String getOesAuditModel(String name, Long timestamp, AuditEventType eventType) {
+    OesAuditModel oesAuditModel = new OesAuditModel();
+    Map<String, Object> date = new HashMap<>();
+    date.put("userName", name);
+    date.put("timestamp", timestamp);
+    oesAuditModel.setEventId(UUID.randomUUID().toString());
+    oesAuditModel.setAuditData(date);
+    oesAuditModel.setEventType(eventType);
+    String model = gson.toJson(oesAuditModel, OesAuditModel.class);
+    log.debug("model: {}", model);
```
Comment (rahul-chekuri) [inline gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/AuthenticationAuditListener.java:123]: any sensitive data is printed here?

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/UserActivityAuditListener.java [hunk 2]
```diff
@@ -35,10 +40,12 @@
 @EnableAsync
 public class UserActivityAuditListener implements ApplicationListener {
 
-  // @Autowired private AuditHandler auditHandler;
```
Comment (rahul-chekuri) [inline gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/UserActivityAuditListener.java:38]: remove commented-out code.

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/UserActivityAuditListener.java [hunk 3]
```diff
@@ -56,18 +63,33 @@ public void onApplicationEvent(ApplicationEvent event) {
...
+  private String getOesAuditModel(Map<String, Object> auditData) {
+    OesAuditModel oesAuditModel = new OesAuditModel();
+    Map<String, Object> date = new HashMap<>();
+    date.put("userName", auditData.get("userName"));
+    date.put("timestamp", auditData.get("timestamp"));
+    oesAuditModel.setEventId(UUID.randomUUID().toString());
+    oesAuditModel.setAuditData(date);
+    oesAuditModel.setEventType(AuditEventType.USER_ACTIVITY_AUDIT);
+    String model = gson.toJson(oesAuditModel, OesAuditModel.class);
+    log.info("model: {}", model);
```
Comment (rahul-chekuri) [inline gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/UserActivityAuditListener.java:89]: change log level to debug or trace

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/constant/CamelEndpointConstant.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2015 OpsMx, Inc.
```
Comment (rahul-chekuri) [inline gate-web/src/main/java/com/opsmx/spinnaker/gate/constant/CamelEndpointConstant.java:2]: Copyright year is wrong

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/AuthenticationAuditListener.java (hunk 1 above)
Comment (sudhakaropsmx) [reply gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/AuthenticationAuditListener.java:123]: no @rahul-chekuri

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/constant/CamelEndpointConstant.java (hunk 4 above)
Comment (sudhakaropsmx) [reply gate-web/src/main/java/com/opsmx/spinnaker/gate/constant/CamelEndpointConstant.java:2]: Updated the Year across all files

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/UserActivityAuditListener.java (hunk 3 above)
Comment (sudhakaropsmx) [reply gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/UserActivityAuditListener.java]: changed to debug mode

**Review by rahul-chekuri — APPROVED** [review]

**Pull request merged**